### PR TITLE
fix: skip adding invalid if not exists to cmd topic

### PR DESCRIPTION
**Pull request by wcarlson5**
changes @spena suggested instead

### Testing done 
_Describe the testing strategy. Unit and integration tests are expected for any behavior changes._

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

File: ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/computation/DistributingExecutor.java [hunk 1]
```diff
@@ -123,6 +154,15 @@ public StatementExecutorResponse execute(
       );
     }
 
+    if (shouldGeneratePlan(executionContext, statement).isPresent()) {
+      return StatementExecutorResponse.handled(Optional.of(
+          new WarningEntity(injected.getStatementText(),
+          "Cannot add `"
+              + shouldGeneratePlan(executionContext, statement).get()
+              + "`: An entity with the same name already exists."
```
Comment (spena) [inline ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/computation/DistributingExecutor.java:162]: Could you edit the message so it is similar to the previous message? The below message is thrown in Error or Warnings currently.

```
String.format("Cannot add table %s: A %s with the same name "
                    + "already exists.",
                sourceName, sourceType.toLowerCase()));
```

File: ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/computation/DistributingExecutor.java [hunk 2]
```diff
@@ -91,6 +98,30 @@ public DistributingExecutor(
         Objects.requireNonNull(commandRunnerWarning, "commandRunnerWarning");
   }
 
+  private Optional<SourceName> shouldGeneratePlan(final KsqlExecutionContext executionContext,
```
Comment (spena) [inline ksqldb-rest-app/src/main/java/io/confluent/ksql/rest/server/computation/DistributingExecutor.java:101]: Rename the method for a better meaning now it belongs to the DistributingExecutor. Perhaps `getCreateSourceNameIfExists()`?

**Review by spena — COMMENTED** [review]
Thanks @wcarlson5 . This seems much simpler. Could you add unit tests for this? Also, add some integration tests. Perhaps `KsqlResourceTest`? Or `EndToEndIntegrationTest` or a QTT?

**Comment by wcarlson5** [comment]
@spena I added a `RecoveryTest` and some unit test. Neither `EndToEndIntegrationTest` or a QTT use the `DistributingExecutor`

**Comment by wcarlson5** [comment]
I just finished running through some manual testing and was able to verify the msg and that the query doesn't vanish on restart

**Review by spena — APPROVED** [review]
Awesome, this looks better. Thanks.

**Pull request merged**